float64[] trMx
float64[] trMy
float64 detM
float64[] u_cc_x
float64[] u_cc_y
float64[] u_ca_x
float64[] u_ca_y
float64[] u_fc_x
float64[] u_fc_y
